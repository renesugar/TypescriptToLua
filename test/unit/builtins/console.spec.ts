import * as util from "../../util";

const compilerOptions = { lib: ["lib.esnext.d.ts", "lib.dom.d.ts"] };

test.each([
    "console.log()",
    'console.log("Hello")',
    'console.log("Hello %s", "there")',
    'console.log("Hello %%s", "there")',
    'console.log("Hello", "There")',
])("console.log (%p)", code => {
    util.testFunction(code).setOptions(compilerOptions).expectLuaToMatchSnapshot();
});

test.each([
    "console.info()",
    'console.info("Hello")',
    'console.info("Hello %s", "there")',
    'console.info("Hello %%s", "there")',
    'console.info("Hello", "There")',
])("console.info (%p)", code => {
    util.testFunction(code).setOptions(compilerOptions).expectLuaToMatchSnapshot();
});

test.each([
    "console.error()",
    'console.error("Hello")',
    'console.error("Hello %s", "there")',
    'console.error("Hello %%s", "there")',
    'console.error("Hello", "There")',
])("console.error (%p)", code => {
    util.testFunction(code).setOptions(compilerOptions).expectLuaToMatchSnapshot();
});

test.each([
    "console.warn()",
    'console.warn("Hello")',
    'console.warn("Hello %s", "there")',
    'console.warn("Hello %%s", "there")',
    'console.warn("Hello", "There")',
])("console.warn (%p)", code => {
    util.testFunction(code).setOptions(compilerOptions).expectLuaToMatchSnapshot();
});

test.each([
    "console.trace()",
    'console.trace("message")',
    'console.trace("Hello %s", "there")',
    'console.trace("Hello %%s", "there")',
    'console.trace("Hello", "there")',
])("console.trace (%p)", code => {
    util.testFunction(code).setOptions(compilerOptions).expectLuaToMatchSnapshot();
});

test.each([
    "console.assert(false)",
    'console.assert(false, "message")',
    'console.assert(false, "message %s", "info")',
    'console.assert(false, "message %%s", "info")',
    'console.assert(false, "message", "more")',
])("console.assert (%p)", code => {
    util.testFunction(code).setOptions(compilerOptions).expectLuaToMatchSnapshot();
});

test("console.differentiation", () => {
    util.testModule`
        export class Console {
            public test() {
                return 42;
            }
        }

        function test() {
            const console = new Console();
            return console.test();
        }

        export const result = test();
    `
        .setReturnExport("result")
        .expectToMatchJsResult();
});
