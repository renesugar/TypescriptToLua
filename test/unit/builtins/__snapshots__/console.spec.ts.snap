// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`console.assert ("console.assert(false)") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    assert(false)
end
return ____exports"
`;

exports[`console.assert ("console.assert(false, \\"message %%s\\", \\"info\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    assert(
        false,
        string.format(\\"message %%s\\", \\"info\\")
    )
end
return ____exports"
`;

exports[`console.assert ("console.assert(false, \\"message %s\\", \\"info\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    assert(
        false,
        string.format(\\"message %s\\", \\"info\\")
    )
end
return ____exports"
`;

exports[`console.assert ("console.assert(false, \\"message\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    assert(false, \\"message\\")
end
return ____exports"
`;

exports[`console.assert ("console.assert(false, \\"message\\", \\"more\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    assert(false, \\"message\\", \\"more\\")
end
return ____exports"
`;

exports[`console.error ("console.error()") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print()
end
return ____exports"
`;

exports[`console.error ("console.error(\\"Hello %%s\\", \\"there\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(
        string.format(\\"Hello %%s\\", \\"there\\")
    )
end
return ____exports"
`;

exports[`console.error ("console.error(\\"Hello %s\\", \\"there\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(
        string.format(\\"Hello %s\\", \\"there\\")
    )
end
return ____exports"
`;

exports[`console.error ("console.error(\\"Hello\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(\\"Hello\\")
end
return ____exports"
`;

exports[`console.error ("console.error(\\"Hello\\", \\"There\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(\\"Hello\\", \\"There\\")
end
return ____exports"
`;

exports[`console.info ("console.info()") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print()
end
return ____exports"
`;

exports[`console.info ("console.info(\\"Hello %%s\\", \\"there\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(
        string.format(\\"Hello %%s\\", \\"there\\")
    )
end
return ____exports"
`;

exports[`console.info ("console.info(\\"Hello %s\\", \\"there\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(
        string.format(\\"Hello %s\\", \\"there\\")
    )
end
return ____exports"
`;

exports[`console.info ("console.info(\\"Hello\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(\\"Hello\\")
end
return ____exports"
`;

exports[`console.info ("console.info(\\"Hello\\", \\"There\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(\\"Hello\\", \\"There\\")
end
return ____exports"
`;

exports[`console.log ("console.log()") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print()
end
return ____exports"
`;

exports[`console.log ("console.log(\\"Hello %%s\\", \\"there\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(
        string.format(\\"Hello %%s\\", \\"there\\")
    )
end
return ____exports"
`;

exports[`console.log ("console.log(\\"Hello %s\\", \\"there\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(
        string.format(\\"Hello %s\\", \\"there\\")
    )
end
return ____exports"
`;

exports[`console.log ("console.log(\\"Hello\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(\\"Hello\\")
end
return ____exports"
`;

exports[`console.log ("console.log(\\"Hello\\", \\"There\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(\\"Hello\\", \\"There\\")
end
return ____exports"
`;

exports[`console.trace ("console.trace()") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(
        debug.traceback()
    )
end
return ____exports"
`;

exports[`console.trace ("console.trace(\\"Hello %%s\\", \\"there\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(
        debug.traceback(
            string.format(\\"Hello %%s\\", \\"there\\")
        )
    )
end
return ____exports"
`;

exports[`console.trace ("console.trace(\\"Hello %s\\", \\"there\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(
        debug.traceback(
            string.format(\\"Hello %s\\", \\"there\\")
        )
    )
end
return ____exports"
`;

exports[`console.trace ("console.trace(\\"Hello\\", \\"there\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(
        debug.traceback(\\"Hello\\", \\"there\\")
    )
end
return ____exports"
`;

exports[`console.trace ("console.trace(\\"message\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(
        debug.traceback(\\"message\\")
    )
end
return ____exports"
`;

exports[`console.warn ("console.warn()") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print()
end
return ____exports"
`;

exports[`console.warn ("console.warn(\\"Hello %%s\\", \\"there\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(
        string.format(\\"Hello %%s\\", \\"there\\")
    )
end
return ____exports"
`;

exports[`console.warn ("console.warn(\\"Hello %s\\", \\"there\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(
        string.format(\\"Hello %s\\", \\"there\\")
    )
end
return ____exports"
`;

exports[`console.warn ("console.warn(\\"Hello\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(\\"Hello\\")
end
return ____exports"
`;

exports[`console.warn ("console.warn(\\"Hello\\", \\"There\\")") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    print(\\"Hello\\", \\"There\\")
end
return ____exports"
`;
